nextcord/http.py

```diff
@@ -240,151 +707,280 @@ async def request(
             if reason:
                 headers["X-Audit-Log-Reason"] = _uriquote(reason, safe="/ ")
 
-        kwargs["headers"] = headers
+        auth = headers.get("Authorization")
 
-        # Proxy support
-        if self.proxy is not None:
-            kwargs["proxy"] = self.proxy
-        if self.proxy_auth is not None:
-            kwargs["proxy_auth"] = self.proxy_auth
+        # If a global rate limit for this authorization doesn't exist yet, make it.
+        if (global_rate_limit := self._global_rate_limits.get(auth)) is None:
+            global_rate_limit = self._make_global_rate_limit(auth, self._max_global_requests)
 
-        if not self._global_over.is_set():
-            # wait until the global lock is complete
-            await self._global_over.wait()
+        # If a rate limit for this url path doesn't exist yet, make it.
+        if (url_rate_limit := self._get_url_rate_limit(route.method, route, auth)) is None:
+            url_rate_limit = self._make_url_rate_limit(route.method, route, auth)
 
+        max_retry_count = 5
+        rate_limit_path = (
+            route.method,
+            route.bucket,
+            _get_logging_auth(auth),
+        )  # Only use this for logging.
+        ret: Union[Optional[str], dict] = None
         response: Optional[aiohttp.ClientResponse] = None
-        data: Optional[Union[Dict[str, Any], str]] = None
-        await lock.acquire()
-        with MaybeUnlock(lock) as maybe_lock:
-            for tries in range(5):
-                if files:
-                    for f in files:
-                        f.reset(seek=tries)
-
-                if form:
-                    form_data = aiohttp.FormData(quote_fields=False)
-                    for params in form:
-                        form_data.add_field(**params)
-                    kwargs["data"] = form_data
-
-                try:
-                    async with self.__session.request(method, url, **kwargs) as response:
+
+        # If retry_request is False and any of the rate limits are locked, don't continue and raise immediately.
+        if retry_request is False:
```

Comment:
I feel like this is unexpected behavior. Waiting for the rate limit to reset isn't retrying the request?